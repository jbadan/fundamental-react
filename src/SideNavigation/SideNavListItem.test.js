import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { SideNav } from '../';

describe('<SideNavListItem />', () => {
    const sideNavListItem = (
        <SideNav.ListItem
            id='subitem_41'
            name='Item 1'
            url='#' />
    );

    test('create side navigation', () => {
        let component = renderer.create(sideNavListItem);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the SideNav.ListItem component', () => {
            const element = mount(<SideNav.ListItem data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
