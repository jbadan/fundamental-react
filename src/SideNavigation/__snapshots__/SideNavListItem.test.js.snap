// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavListItem /> create side navigation 1`] = `
<li
  className="fd-side-nav__item"
>
  <a
    className="fd-side-nav__link"
    href="#"
    onClick={[Function]}
  >
    Item 1
  </a>
</li>
`;
